frictionless/steps/field/field_pack.py

```diff
@@ -0,0 +1,56 @@
+from ...step import Step
+from ...field import Field
+from ...helpers import iterpack, iterpackdict
+
+
+class field_pack(Step):
+    """Pack field"""
+
+    code = "field-pack"
+
+    def __init__(
+        self,
+        descriptor=None,
+        *,
+        name=None,
+        from_names=None,
+        field_type=None,
+        preserve=False
+    ):
+        self.setinitial("name", name)
+        self.setinitial("fromNames", from_names)
+        self.setinitial("fieldType", field_type)
+        self.setinitial("preserve", preserve)
+        super().__init__(descriptor)
+
+    # Transform
+
+    def transform_resource(self, resource):
+        table = resource.to_petl()
+        name = self.get("name")
+        from_names = self.get("fromNames")
+        field_type = self.get("fieldType")
+        preserve = self.get("preserve")
+        resource.schema.add_field(Field(name=name))
+        if not preserve:
+            for name in from_names:
+                resource.schema.remove_field(name)
+        if field_type == "object":
```

Comment:
Is it possible to use PETL's functions here instead of copying them into our helpers? (similar for merge)

Reply:
PETL doesnot merge and pack function so I wrote it, but I took reference from their code for loops and the flow.